### ISSUE TITLE
spawn ENAMETOOLONG thrown by XMLHttpRequest.send

### ISSUE THREAD

**Issue by veeprox** [google/earthengine-api#37]
Hi There,

We are using the Earth Engine Javascript API in a Node/Express application to make some calls to generate maps and tiles. I've noticed that if you make a synchronous request, i.e.

```
let boundaryGeo = ee.Geometry(boundary);
let envelopeGeo = boundaryGeo.bounds().getInfo(); //this line fails
let envelopeGeoString = JSON.stringify(envelopeGeo);
```

you get an error message `Error: spawn ENAMETOOLONG` sometimes because Node is trying to create an XML file similar to `.node-xmlhttprequest-sync-55084` behind the scenes to marshal this off onto a child process.

This usually happens if you are making requests that involve polygons or multipolygons that are very large. The Polygon I am using for example is about 50,000 characters long when serialised as a GeoJSON.

Adding a callback makes this work fine; i.e.

```
let boundaryGeo = ee.Geometry(boundary);
boundaryGeo.bounds().getInfo((envelopeGeo) => {
  //works fine
  let envelopeGeoString = JSON.stringify(envelopeGeo);
}); 

```

I'm happy for `getInfo(...)` or `startProcessing(...)` to run synchronously when I'm running server side in Node as opposed to the browser. I don't really want to mix my async/await pattern with callbacks in this NodeJS function if I can help it.

Thoughts?

edit: added some extra info

**Comment by veeprox** [comment]
Here is the offending polygon - https://gist.github.com/veeprox/135096a71c145c59581dafed28b900da

**Comment by donmccurdy** [comment]
> I'm happy for getInfo(...) or startProcessing(...) to run synchronously when I'm running server side in Node as opposed to the browser.

An important note here is that Node.js does not natively have synchronous requests, and we're going to some lengths to run `getInfo()` synchronously — specifically, the server process _stops_, and no other visitors to your application are served while waiting for this response. If you're building an application designed to handle multiple concurrent users, it is **highly** recommended to use asynchronous requests with callbacks.

That said, I'm surprised an error is thrown — it does seem like a bug. In what context are you running Express? If locally, what OS? If on a remote server, what provider?

Thanks!

**Comment by veeprox** [comment]
Hi Don, thanks for the reply.

It's been a while since I worked on this one but the issue was occuring on my local development machine (Windows 10, Node v9.3). It doesn't matter if I run the NodeJS app through Visual Studio or run it from the command line. It's interesting though that I dont believe the error happens if you use a simple polygon - it seems to happen with big shapes like I linked in the gist.

I guess it's not a big deal to use the callbacks - we were using Typescript and async/await to avoid callbacks where possible. If I end up working on this code again i'll just wrap the `getInfo(callback);` in a method that returns a promise so i can still await the a function.
